### PR TITLE
Add groups for phone use permissions.

### PR DESCRIPTION
**Pull request by Tomin1**
Add sailfish-calls and sailfish-messages groups for calls and SMSs respectively. Defaults to users having those permissions. Also clarify group definitions for existing groups and clean up hardware specific groups from USER_GROUPS.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,19 +25,28 @@ Groups that this package defines are listed below.
...
 - sailfish-datetime
-  - For processes that need to be able to modify clock settings.
+  - For users that need to be able to modify clock settings.
 - timed
-  - For timed process, required for changing system default timezone
+  - For users and timed process, required to change system default
+    timezone.
+- sailfish-actdead
+  - For sailfish-actdead user.
+- oneshot
```
Comment (Tomin1) [inline README.md]: Is this actually used? To me it looks like it's not anymore.

**Review by mkenttala — APPROVED** [review]
Fine by me

File: rpm/sailfish-setup.spec [hunk 2]
```diff
@@ -47,6 +50,10 @@ groupadd -rf sailfish-datetime || :
 
 groupadd -rf timed || :
 
+groupadd -rf sailfish-calls || :
```
Comment (rainemak) [inline rpm/sailfish-setup.spec]: Maybe this should be called "sailfish-phone" or "sailfish-voicecall"

File: README.md (hunk 1 above)
Comment (rainemak) [reply README.md]: Same here. None of the dirs afais has oneshot group: 

> find / -group oneshot

Comment (rainemak) [reply README.md]: Go ahead and nuke this.

File: rpm/sailfish-setup.spec (hunk 2 above)
Comment (Tomin1) [reply rpm/sailfish-setup.spec]: I chose the sailfish-phone, although I know it's not that great name to limit things on lower level but it mirrors sailfish-messages nicely. sailfish-voicecall would mean we'd need to call sailfish-messages to sailfish-sms or something but they wouldn't have connection to the applications. Hard to decide although it's such a small thing.

File: group.ids [hunk 3]
```diff
@@ -1,2 +1,2 @@
 SYSTEM_GROUPS sailfish-system,sailfish-datetime,timed,system,ssu
-USER_GROUPS video,input,audio,users,oneshot,sailfish-alarms,bluetooth,graphics,camera,media,media_rw,mtp,inet
+USER_GROUPS video,input,audio,users,sailfish-alarms,sailfish-phone,sailfish-messages
```
Comment (rainemak) [inline group.ids]: If @mkosola thinks that bluetooth... can be now go then I'm ok with this.

Comment (mkosola) [reply group.ids]: @rainemak bluetooth group is added here https://git.sailfishos.org/mer-core/bluez5/blob/master/rpm/bluez5.spec#L237

Comment (jusa) [reply group.ids]: that bluez5 line just adds nemo user to bluetooth group.

So we definitely need the bluetooth group here.

Bluez5 probably should install file to /usr/lib/droid/hw-group.d with line "bluetooth".

Comment (Tomin1) [reply group.ids]: Ok, so bluetooth is needed for now at least. Well, what ensures that bluetooth group is there when user is added to these groups? Adding happens when user session is started or when user is created. That file there sounds like a more robust approach.

I'd prefer not to add such things here that sailfish-setup doesn't either create or depend on but as we don't anymore create users on image creation groups created elsewhere probably work here too.

Comment (rainemak) [reply group.ids]: @jusa  but I'm seeing "bluetooth" line in the /usr/lib/droid/hw-group.d/default.list

Comment (Tomin1) [reply group.ids]: Well, added bluetooth back. At least for the time being. Are the other groups okay though? They are in the adaptation file but so is audio.

Comment (Tomin1) [reply group.ids]: Also like, my reasoning for removing these hardware specific groups was that they weren't part of this until this file was added so they shouldn't be necessary. I guess audio and bluetooth are a bit special as they are also needed by native applications.

**Comment by rainemak** [comment]
Let's see what happens. Approving as well.

**Review by rainemak — APPROVED** [review]
LGTM

**Pull request merged**